tensorboard/webapp/header/plugin_selector_component.ts

```diff
@@ -41,8 +40,9 @@ export class PluginSelectorComponent {
     return this.activePlugins.findIndex(({id}) => id === this.selectedPlugin);
   }
 
-  onActivePluginSelectionChanged({index}: MatTabChangeEvent) {
-    this.onPluginSelectionChanged.emit(this.activePlugins[index].id);
+  onActivePluginSelection(event: Event, pluginId: PluginId) {
+    event.stopPropagation();
+    this.onPluginSelectionChanged.emit(pluginId);
```

Comment:
Do I read correctly that we no longer emit `onPluginSelectionChanged`
when the plugin changes due to a hash change event, and that that’s
acceptable because this is only used to dispatch a `changePlugin`
action, and in the case of a hash change event a different container
will fire that action?

Reply:
Correct but clarification is required.

The fact that we fire `onPluginSelectionChanged ` when we set the index programmatically was not a desired but okay behavior. This has almost nothing to do with the hash since literally grab the value from the store and set it as the value to the component.

> when the plugin changes due to a hash change event

This is handled by some other module called deep linker which is responsible for listening to hash change and notify store accordingly. If that event happens to change the store state, then plugin_selector_component will receive the new value and trigger `onPluginSelectionChanged`.

Reply:
Yep; makes sense. Thanks!